module/VuFind/src/VuFind/View/Helper/Root/CookieConsent.php

```diff
@@ -0,0 +1,441 @@
+<?php
+/**
+ * CookieConsent view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\View\Helper\Root;
+
+use VuFind\Cookie\CookieManager;
+use VuFind\I18n\Translator\TranslatorAwareInterface;
+use VuFind\I18n\Translator\TranslatorAwareTrait;
+
+/**
+ * CookieConsent view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class CookieConsent extends \Laminas\View\Helper\AbstractHelper
+    implements TranslatorAwareInterface
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * Main configuration
+     *
+     * @var array
+     */
+    protected $config;
+
+    /**
+     * Cookie consent configuration
+     *
+     * @var array
+     */
+    protected $consentConfig;
+
+    /**
+     * Cookie manager
+     *
+     * @var CookieManager
+     */
+    protected $cookieManager;
+
+    /**
+     * Consent cookie name
+     *
+     * @var string
+     */
+    protected $consentCookieName;
+
+    /**
+     * Consent cookie expiration time
+     *
+     * @var int
+     */
+    protected $consentCookieExpiration;
+
+    /**
+     * Server name
+     *
+     * @var string
+     */
+    protected $hostName = null;
+
+    /**
+     * Constructor
+     *
+     * @param array         $config        Main configuration
+     * @param array         $consentConfig Cookie consent configuration
+     * @param CookieManager $cookieManager Cookie manager
+     */
+    public function __construct(
+        array $config,
+        array $consentConfig,
+        CookieManager $cookieManager
+    ) {
+        $this->config = $config;
+        $this->consentConfig = $consentConfig;
+        $this->cookieManager = $cookieManager;
+        $this->consentCookieName = $this->consentConfig['CookieName'] ?? 'cc_cookie';
+        $this->consentCookieExpiration
+            = $this->consentConfig['CookieExpiration'] ?? 182;
```

Comment:
Why 182? Just seems like an oddly specific magic number...

Reply:
Half a year rounded down. That's the default for the cookieconsent module, but I'm not sure where it comes from. I'll let you know if I find a clarification for it.

Reply:
Ahh, of course. Maybe a comment `// half a year` at the end of the line would be enough to satisfy curiosity. :-)